dev/src/dev/migrate.clj

```diff
@@ -116,27 +118,39 @@
 (defn- change->sql
   [change sql-generator-factory database]
   {:forward  (stmts-to-sql (.generateStatements change database) sql-generator-factory database)
-   :rollback (stmts-to-sql (.generateRollbackStatements change database) sql-generator-factory database)})
-
-(defn migration-sql-by-id
-  "Get the sql statements for a specific migration ID.
-    (migration-sql-by-id \"v51.2024-06-12T18:53:02\")
+   :rollback (try (stmts-to-sql (.generateRollbackStatements change database) sql-generator-factory database)
+                  (catch RollbackImpossibleException e
+                    (str "Rollback impossible " e)))})
+
+(defn- liquibase-database [db-type]
+  (case db-type
+    :postgres (PostgresDatabase.)
+    :mysql    (MySQLDatabase.)
+    :mariadb  (MariaDBDatabase.)
+    :h2       (H2Database.)))
+
+(mu/defn migration-sql-by-id
+  "Get the sql statements for a specific migration ID and DB type. If no DB type is provided, it will use the current
+   application DB type.
+    (migration-sql-by-id \"v51.2024-06-12T18:53:02\" :postgres)
     ;; =>
       {:forward \"DROP INDEX public.idx_user_id_device_id;\",
        :rollback \"CREATE INDEX idx_user_id_device_id ON public.login_history(session_id, device_id);\"}"
-  [id]
-  (t2/with-connection [conn]
-    (liquibase/with-liquibase [^Liquibase liquibase conn]
-      (let [database            (.getDatabase liquibase)
-            change-log-iterator (ChangeLogIterator. (.getDatabaseChangeLog liquibase) (into-array ChangeSetFilter []))
-            list-visistor       (ListVisitor.)
-            runtime-env         (RuntimeEnvironment. database (Contexts.) nil)
-            _                   (.run change-log-iterator list-visistor runtime-env)
-            change-set          (first (filter #(= id (.getId %))(.getSeenChangeSets list-visistor)))
-            sql-generator-factory (SqlGeneratorFactory/getInstance)]
-        (reduce (fn [acc data]
+  ([id]
+   (migration-sql-by-id id (mdb/db-type)))
```

Comment:
Instead of using `(.getDatabase liquibase)` we now use `mdb/db-type` to get the keyword, and then `liquibase-database` to get a liquibase `Database` instance.